src/WrappedMToken.sol

```diff
@@ -561,16 +561,13 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
      * @return wrapped_   The amount of wM minted.
      */
     function _wrap(address account_, address recipient_, uint240 amount_) internal returns (uint240 wrapped_) {
-        uint240 startingBalance_ = _mBalanceOf(address(this));
-
         // NOTE: The behavior of `IMTokenLike.transferFrom` is known, so its return can be ignored.
         IMTokenLike(mToken).transferFrom(account_, address(this), amount_);
 
         // NOTE: When this WrappedMToken contract is earning, any amount of M sent to it is converted to a principal
         //       amount at the MToken contract, which when represented as a present amount, may be a rounding error
-        //       amount less than `amount_`. In order to capture the real increase in M, the difference between the
-        //       starting and ending M balance is minted as WrappedM token.
-        _mint(recipient_, wrapped_ = _mBalanceOf(address(this)) - startingBalance_);
```

Comment:
While this fix does solve the problem of the user receiving a rounded down amount, it can make the wrapped M token slightly insolvent because it is essentially eating the deficit. This may not be a big deal from a business perspective, but it's possible to show that a roundtrip wrap/unwrap operation fails due to this discrepancy when there are no other tokens in the contract. See the `testFuzz_wrap_unwrap_roundtripFull` test in this branch: https://github.com/m0-foundation/wrapped-m-token/tree/wrap/unwrap_test

Reply:
Yep. This the known trade off: rounding in favour of the protocol vs rounding in favour of the user. And it is why the deployed version does not have these changes. To manage solvency, we need to hope that:
- there is enough excess to cover all the rounding loss the wrapper suffers due to rounding in favour of the users, and/or
- one or more accounts hold off of unwrapping and have enough balance to cover all the rounding loss the wrapper suffers due to rounding in favour of the users